packages/udaru-core/index.js

```diff
@@ -20,78 +21,80 @@ function buildUdaruCore (dbPool, config) {
 
   return {
     config,
-    getUserOrganizationId: userOps.getUserOrganizationId,
+    getUserOrganizationId: hooks.wrap('user:getUserOrganizationId', userOps.getUserOrganizationId),
 
     db: {
       close: db.shutdown
     },
 
+    addHook: hooks.addHook,
+
     authorize: {
-      isUserAuthorized: authorizeOps.isUserAuthorized,
-      listActions: authorizeOps.listAuthorizations,
-      listAuthorizationsOnResources: authorizeOps.listAuthorizationsOnResources
+      isUserAuthorized: hooks.wrap('authorize:isUserAuthorized', authorizeOps.isUserAuthorized),
+      listActions: hooks.wrap('authorize:listActions', authorizeOps.listAuthorizations),
+      listAuthorizationsOnResources: hooks.wrap('authorize:listAuthorizationsOnResources', authorizeOps.listAuthorizationsOnResources)
     },
 
     organizations: {
-      list: organizationOps.list,
-      create: organizationOps.create,
-      read: organizationOps.readById,
-      delete: organizationOps.deleteById,
-      update: organizationOps.update,
-      addPolicies: organizationOps.addOrganizationPolicies,
-      replacePolicies: organizationOps.replaceOrganizationPolicies,
-      deletePolicies: organizationOps.deleteOrganizationAttachedPolicies,
-      deletePolicy: organizationOps.deleteOrganizationAttachedPolicy
+      list: hooks.wrap('organization:list', organizationOps.list),
+      create: hooks.wrap('organization:create', organizationOps.create),
+      read: hooks.wrap('organization:read', organizationOps.readById),
+      delete: hooks.wrap('organization:delete', organizationOps.deleteById),
+      update: hooks.wrap('organization:update', organizationOps.update),
+      addPolicies: hooks.wrap('organization:addPolicies', organizationOps.addOrganizationPolicies),
+      replacePolicies: hooks.wrap('organization:replacePolicies', organizationOps.replaceOrganizationPolicies),
+      deletePolicies: hooks.wrap('organization:deletePolicies', organizationOps.deleteOrganizationAttachedPolicies),
+      deletePolicy: hooks.wrap('organization:deletePolicy', organizationOps.deleteOrganizationAttachedPolicy)
```

Comment:
so does this mean that we can have hooks on all these functions? I wouldn't of seen the need to wrap read and list commands for example, but I guess if we get them for free, why not..

Reply:
I don't see many use cases as well, but who knows. Might be useful for security monitoring.
Anyway each wrapping has small cost on bootstraping. 
When the handlers list is empty each call will result in additional function invoked with iteration on a empty array. So it shouldn't impact performances at all. Especially when these methods are not really used. :)